docs_src/security/V3Migration.md

```diff
@@ -1,3 +1,118 @@
+that can be used to request a JWT from Vault.
+
+Externally-originated requests are performed similarly to how they were done before:
+provide the JWT in the `Authorization` header
+and direct the request at the API gateway with a path prefix denoting the desired service.
+The key difference is in obtaining the JWT.
+In EdgeX 2.0, the client simply generated the JWT using its private key.
+In EdgeX 3.0, obtaining a JWT is a two-step process.
+First, authenticate to the EdgeX secret store (Vault) to obtain a secret store token.
+Second, exchange the secret store token for a JWT.
+This process is described in detail in the
+[authenticating chapter](https://docs.edgexfoundry.org/3.0/security/Ch-Authenticating/)
+of the EdgeX documentation.
+Due to these changes, the `secrets-config proxy jwt` helper command has been removed.
+This same chapter also explains that, similar to Kong,
+Vault has an extensible authentication mechanism,
+although only username/password (with a randomized strong password)
+is enabled out of the box.
+
+As was before, all requests (with the exception of a passthrough for Vault authentication)
+are checked at the API gateway prior to forwarding to the backend service for fulfillment.
+
+## Microservice-level Breaking Authentication Changes
+
+EdgeX microservices in EdgeX 3.0 will now require authentication on a per-route basis,
+even for requests that originate behind the API gateway.
+Peer-to-peer service requests (such as a device service calling core-metadata,
+or core-command forwarding a request to a device service) are authenticated automatically.
+This new behavior may create compatibility issues for custom components
+that worked fine in EdgeX 2.0 that may suddenly experience authentication failures in EdgeX 3.0.
+This new behavior may also create issues for 3rd party components,
+such as the eKuiper rules engine,
+because of its ability to issue ad-hoc HTTP requests in response to certain events.
+
+To revert to legacy EdgeX 2.0 behavior--no authentication at the microservice level--
+set the environment variable `EDGEX_DISABLE_JWT_VALIDATION` to `true`.
+JWT validation must be disabled on a per-microservice basis.
+This will not stop EdgeX microservices from sending JWT's
+to peer EdgeX microservices--it will only disable validation on the receiving side,
+allowing unauthenticated requests.
+
+For sending JWTs, custom EdgeX services have two basic choices.
+The first is to use one of the pre-built service clients in `go-mod-core-contracts`.
+The other is to to use the `GetSelfJWT()` method of the `SecretProviderExt` interface.
+The
+[authenticating chapter](https://docs.edgexfoundry.org/3.0/security/Ch-Authenticating/)
+of the EdgeX documentation
+explains in greater detail how to use these two methods.
+
+## Breaking Changes to API Gateway TLS Configuration
+
+Some minor changes have been made to the `secrets-config proxy tls` command.
+For starters, the `--snis` argument is no longer supported--the
```

Comment:
```suggestion
For starters, the `--snis` argument is no longer supported. The
```

Reply:
Would be easier to read as a bulleted list.

Reply:
Fixed.